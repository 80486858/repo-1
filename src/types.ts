export type AssetDiff = {
  name: string
  new: {
    size: number
    gzipSize: number | null
  }
  old: {
    size: number
    gzipSize: number | null
  }
  diff: number
  diffPercentage: number
}

export type Sizes = {
  size: number
  gzipSize: number | null
}

export type WebpackStatsDiff = {
  added: AssetDiff[]
  removed: AssetDiff[]
  bigger: AssetDiff[]
  smaller: AssetDiff[]
  unchanged: AssetDiff[]
  total: AssetDiff
}
