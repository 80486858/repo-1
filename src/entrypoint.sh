#! /bin/bash

if [ -z "${DASHLANE_CLI_TIMESTAMP}" ]
then
    DASHLANE_CLI_TIMESTAMP=$(date -d '1 day ago' +%s000)
fi

while true
do
    DASHLANE_CLI_RESULT=$(dcli t l --start $DASHLANE_CLI_TIMESTAMP --end now)
    echo $DASHLANE_CLI_RESULT | /opt/fluent-bit/bin/fluent-bit -c $DASHLANE_CLI_FLUENTBIT_CONF -q
    DASHLANE_CLI_TIMESTAMP=$(date +%s000)
    sleep ${DASHLANE_CLI_RUN_DELAY:-1800}
done