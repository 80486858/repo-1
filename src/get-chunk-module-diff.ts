import type {StatsCompilation} from 'webpack'
import {chunkModuleNameToSizeMap} from './name-to-size-map'
import type {WebpackStatsDiff} from './types'
import {webpackStatsDiff} from './webpack-stats-diff'

export function getChunkModuleDiff(
  oldStats: Pick<StatsCompilation, 'chunks'>,
  newStats: Pick<StatsCompilation, 'chunks'>
): WebpackStatsDiff | null {
  if (!oldStats.chunks || !newStats.chunks) {
    return null
  }
  return webpackStatsDiff(
    chunkModuleNameToSizeMap(oldStats.chunks),
    chunkModuleNameToSizeMap(newStats.chunks)
  )
}
