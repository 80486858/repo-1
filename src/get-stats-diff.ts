import type {StatsCompilation} from 'webpack'
import {nameToSizeMap} from './name-to-size-map'
import type {WebpackStatsDiff} from './types'
import {webpackStatsDiff} from './webpack-stats-diff'

export function getStatsDiff(
  oldAssetStats: Pick<StatsCompilation, 'assets'>,
  newAssetStats: Pick<StatsCompilation, 'assets'>
): WebpackStatsDiff {
  return webpackStatsDiff(
    nameToSizeMap(oldAssetStats.assets),
    nameToSizeMap(newAssetStats.assets)
  )
}
