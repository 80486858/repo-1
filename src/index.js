var BN = require('bn.js');
var stripHexPrefix = require('strip-hex-prefix');

/**
 * Returns a BN object, converts a number value to a BN
 * @param {String|Number|Object} `arg` input a string number, hex string number, number, BigNumber or BN object
 * @return {Object} `output` BN object of the number
 * @throws if the argument is not an array, object that isn't a bignumber, not a string number or number
 */
module.exports = function numberToBN(arg) {
  if (typeof arg === 'string' || typeof arg === 'number') {
    var multiplier = new BN(1); // eslint-disable-line
    var stringArg = stripHexPrefix(String(arg).toLowerCase().trim()); // eslint-disable-line
    if (stringArg.substr(0, 1) === '-') {
      stringArg = stripHexPrefix(stringArg.slice(1));
      multiplier = new BN(-1);
    }

    if (stringArg.match(/^[0-9][a-fA-F]+$/) || stringArg.match(/^[a-fA-F]+$/)) {
      return new BN(stringArg, 16).mul(multiplier);
    } else if (stringArg.match(/^-?[0-9]+$/) || stringArg === '') {
      return new BN(stringArg, 10).mul(multiplier);
    }
  } else if (typeof arg === 'object' && arg.toString && (!arg.pop && !arg.push)) {
    if (arg.toString(10).match(/^-?[0-9]+$/) && (arg.mul || arg.dividedToIntegerBy)) {
      return new BN(arg.toString(10));
    }
  }

  throw new Error('[number-to-bn] while converting number ' + JSON.stringify(arg) + ' to BN.js instance, error: invalid number value. Value must be an integer, hex string, BN or BigNumber instance. Note, decimals are not supported.');
}
