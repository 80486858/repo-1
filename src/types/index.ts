type Mode = '100644' | '100755' | '040000' | '160000' | '120000' | undefined

export type Tree = {
  path?: string | undefined
  mode?: Mode
  type?: 'blob' | 'commit' | 'tree' | undefined
  sha?: string | null | undefined
  content?: string | undefined
}[]
