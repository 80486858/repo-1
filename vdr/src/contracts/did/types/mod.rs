pub mod did;
pub mod did_doc;
pub mod did_doc_attribute;
pub mod did_doc_builder;
pub mod did_events;
