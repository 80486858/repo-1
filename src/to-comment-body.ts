import {printAssetTablesByGroup, printTotalAssetTable} from './print-markdown'
import {WebpackStatsDiff} from './get-stats-diff'

export function getIdentifierComment(key: string): string {
  return `<!--- bundlestats-action-comment${key ? ` key:${key}` : ''} --->`
}

export function getCommentBody(
  statsDiff: WebpackStatsDiff,
  title: string
): string {
  return `
# Bundle Stats${title ? `-${title}` : ''}

Hey there, this message comes from a github action that helps you and reviewers to understand how these changes affect the size of this project's bundle.

As this PR is updated, I'll keep you updated on how the bundle size is impacted.

${printTotalAssetTable(statsDiff)}

<details>
  <summary>View detailed bundle breakdown</summary>

  <div>

    ${printAssetTablesByGroup(statsDiff)}

  </div>
</details>

${getIdentifierComment(title)}
`
}
