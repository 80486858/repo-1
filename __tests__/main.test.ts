import {expect, test} from '@jest/globals'
import {getStatsDiff} from '../src/get-stats-diff'
import {getChunkModuleDiff} from '../src/get-chunk-module-diff'
import {getCommentBody} from '../src/to-comment-body'
import {
  printAssetTablesByGroup,
  printChunkModulesTable,
  printTotalAssetTable
} from '../src/print-markdown'
import {AssetDiff} from '../src/types'

test('Shows stats when files are removed', () => {
  const statsDiff = getStatsDiff(
    require('./__mocks__/old-stats-assets.json'),
    require('./__mocks__/new-stats-assets.json')
  )

  expect(printTotalAssetTable(statsDiff)).toMatchSnapshot()
  expect(printAssetTablesByGroup(statsDiff)).toMatchSnapshot()
})

test('Shows stats when files are added', () => {
  const statsDiff = getStatsDiff(
    require('./__mocks__/new-stats-assets.json'),
    require('./__mocks__/old-stats-assets.json')
  )

  expect(printTotalAssetTable(statsDiff)).toMatchSnapshot()
  expect(printAssetTablesByGroup(statsDiff)).toMatchSnapshot()
})

test('Shows stats when files are unchanged', () => {
  const statsDiff = getStatsDiff(
    require('./__mocks__/old-stats-assets.json'),
    require('./__mocks__/old-stats-assets.json')
  )

  expect(printTotalAssetTable(statsDiff)).toMatchSnapshot()
  expect(printAssetTablesByGroup(statsDiff)).toMatchSnapshot()
})

test('computes the correct module diff information', () => {
  const statsDiff = getChunkModuleDiff(
    require('./__mocks__/old-stats-with-chunks.json'),
    require('./__mocks__/new-stats-with-chunks.json')
  )

  expect(statsDiff?.added).toContainEqual({
    name: './src/client/this-file-was-added.ts',
    diff: 1496,
    diffPercentage: Infinity,
    new: {size: 1496, gzipSize: NaN},
    old: {size: 0, gzipSize: 0}
  } as AssetDiff)
  expect(statsDiff?.bigger).toContainEqual({
    name: './src/client/this-file-grew-larger.tsx',
    diff: 200,
    diffPercentage: 35.58719,
    new: {size: 762, gzipSize: NaN},
    old: {size: 562, gzipSize: NaN}
  } as AssetDiff)
  expect(statsDiff?.smaller).toContainEqual({
    name: './src/client/helpers/this-file-grew-smaller.ts',
    diff: -200,
    diffPercentage: -7.94281,
    new: {size: 2318, gzipSize: NaN},
    old: {size: 2518, gzipSize: NaN}
  } as AssetDiff)
  expect(statsDiff?.removed).toContainEqual({
    name: './src/client/this-file-will-be-deleted.ts',
    diff: -1496,
    diffPercentage: -100,
    new: {size: 0, gzipSize: 0},
    old: {size: 1496, gzipSize: NaN}
  } as AssetDiff)
  expect(statsDiff?.total.new).toEqual(statsDiff?.total.old)
  expect(statsDiff?.total.diff).toEqual(0)
  expect(statsDiff?.total.diffPercentage).toEqual(0)
})

test('displays module information when files are added/removed/changed', () => {
  const statsDiff = getChunkModuleDiff(
    require('./__mocks__/old-stats-with-chunks.json'),
    require('./__mocks__/new-stats-with-chunks.json')
  )

  expect(printChunkModulesTable(statsDiff)).toMatchSnapshot()
})

test('displays no module information when unchanged', () => {
  const statsDiff = getChunkModuleDiff(
    require('./__mocks__/old-stats-with-chunks.json'),
    require('./__mocks__/old-stats-with-chunks.json')
  )

  expect(printChunkModulesTable(statsDiff)).toMatchSnapshot()
})

test('does not display module information when it does not exist', () => {
  const statsDiff = getChunkModuleDiff(
    require('./__mocks__/old-stats-assets.json'),
    require('./__mocks__/old-stats-assets.json')
  )

  expect(printChunkModulesTable(statsDiff)).toMatchSnapshot()
})
