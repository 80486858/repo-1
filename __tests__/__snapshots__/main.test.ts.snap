// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows stats when files are added 1`] = `
"**Total**

Files count | Type | Total bundle size | % Changed
----------- | ---- | ----------------- | ---------
2 -> 7 | bundled<br />gzip | 1.29 MB -> 1.34 MB (+53.65 KB)<br />N/A -> 386.44 KB | +4.07%"
`;

exports[`Shows stats when files are added 2`] = `
"**Added**

Asset | Type | File Size | % Changed
----- | ---- | --------- | ---------
296.chunk.js | bundled<br />gzip | 0 B -> 124.57 KB (+124.57 KB)<br />0 B -> 35.05 KB | -
288.chunk.js | bundled<br />gzip | 0 B -> 57.24 KB (+57.24 KB)<br />0 B -> 16.33 KB | -
920.chunk.js | bundled<br />gzip | 0 B -> 54.98 KB (+54.98 KB)<br />0 B -> 17.08 KB | -
912.chunk.js | bundled<br />gzip | 0 B -> 44.37 KB (+44.37 KB)<br />0 B -> 14.31 KB | -
699.chunk.js | bundled<br />gzip | 0 B -> 26.39 KB (+26.39 KB)<br />0 B -> 6.14 KB | -

**Removed**

No assets were removed

**Bigger**

Asset | Type | File Size | % Changed
----- | ---- | --------- | ---------
manifest.json | bundled<br />gzip | 91 B -> 551 B (+460 B)<br />N/A -> 151 B | +505.49%

**Smaller**

Asset | Type | File Size | % Changed
----- | ---- | --------- | ---------
app.bundle.js | bundled<br />gzip | 1.29 MB -> 1.04 MB (-254.35 KB)<br />N/A -> 297.38 KB | -19.29%

**Unchanged**

No assets were unchanged"
`;

exports[`Shows stats when files are removed 1`] = `
"**Total**

Files count | Type | Total bundle size | % Changed
----------- | ---- | ----------------- | ---------
7 -> 2 | bundled<br />gzip | 1.34 MB -> 1.29 MB (-53.65 KB)<br />386.44 KB -> N/A | -3.91%"
`;

exports[`Shows stats when files are removed 2`] = `
"**Added**

No assets were added

**Removed**

Asset | Type | File Size | % Changed
----- | ---- | --------- | ---------
296.chunk.js | bundled<br />gzip | 124.57 KB -> 0 B (-124.57 KB)<br />35.05 KB -> 0 B | -100%
288.chunk.js | bundled<br />gzip | 57.24 KB -> 0 B (-57.24 KB)<br />16.33 KB -> 0 B | -100%
920.chunk.js | bundled<br />gzip | 54.98 KB -> 0 B (-54.98 KB)<br />17.08 KB -> 0 B | -100%
912.chunk.js | bundled<br />gzip | 44.37 KB -> 0 B (-44.37 KB)<br />14.31 KB -> 0 B | -100%
699.chunk.js | bundled<br />gzip | 26.39 KB -> 0 B (-26.39 KB)<br />6.14 KB -> 0 B | -100%

**Bigger**

Asset | Type | File Size | % Changed
----- | ---- | --------- | ---------
app.bundle.js | bundled<br />gzip | 1.04 MB -> 1.29 MB (+254.35 KB)<br />297.38 KB -> N/A | +23.91%

**Smaller**

Asset | Type | File Size | % Changed
----- | ---- | --------- | ---------
manifest.json | bundled<br />gzip | 551 B -> 91 B (-460 B)<br />151 B -> N/A | -83.48%

**Unchanged**

No assets were unchanged"
`;

exports[`Shows stats when files are unchanged 1`] = `
"**Total**

Files count | Type | Total bundle size | % Changed
----------- | ---- | ----------------- | ---------
7 | bundled<br />gzip | 1.34 MB<br />386.44 KB | 0%"
`;

exports[`Shows stats when files are unchanged 2`] = `
"**Added**

No assets were added

**Removed**

No assets were removed

**Bigger**

No assets were bigger

**Smaller**

No assets were smaller

**Unchanged**

Asset | Type | File Size | % Changed
----- | ---- | --------- | ---------
app.bundle.js | bundled<br />gzip | 1.04 MB<br />297.38 KB | 0%
296.chunk.js | bundled<br />gzip | 124.57 KB<br />35.05 KB | 0%
288.chunk.js | bundled<br />gzip | 57.24 KB<br />16.33 KB | 0%
920.chunk.js | bundled<br />gzip | 54.98 KB<br />17.08 KB | 0%
912.chunk.js | bundled<br />gzip | 44.37 KB<br />14.31 KB | 0%
699.chunk.js | bundled<br />gzip | 26.39 KB<br />6.14 KB | 0%
manifest.json | bundled<br />gzip | 551 B<br />151 B | 0%"
`;

exports[`displays module information when files are added/removed/changed 1`] = `
"
<details>
<summary>**Changeset**</summary>

File | Î”
---- | -
  \`src/client/this/file/has/a/long/name/so-that-it-will-hopefully-wrap.tsx\` | ðŸ†• +2.72 KB
\`src/client/this-file-was-added.ts\` | ðŸ†• +1.46 KB
\`src/client/this-is-another-file-with-a-long-name-to-test-the-table-sizing.tsx\` | ðŸ†• +1.04 KB
\`src/client/helpers/this-file-is-much-larger.ts\` | ðŸ“ˆ +11.66 KB (+865.77%)
\`src/client/this-file-grew-larger.tsx\` | ðŸ“ˆ +200 B (+35.59%)
\`src/client/helpers/this-file-grew-smaller.ts\` | ðŸ“‰ -200 B (-7.94%)
\`src/client/this-file-is-much-smaller.tsx\` | ðŸ“‰ -11.66 KB (-99.92%)
\`src/client/routes.tsx\` | ðŸ”¥ -2.72 KB (-100%)
\`src/client/this-file-will-be-deleted.ts\` | ðŸ”¥ -1.46 KB (-100%)
\`src/client/render-memex.tsx\` | ðŸ”¥ -1.04 KB (-100%)

<details>
<summary>View individual file sizes</summary>

File | Old | New | Î”
---- | --- | --- | -
  \`src/client/this/file/has/a/long/name/so-that-it-will-hopefully-wrap.tsx\` | 0 B | 2.72 KB | ðŸ†• +2.72 KB
\`src/client/this-file-was-added.ts\` | 0 B | 1.46 KB | ðŸ†• +1.46 KB
\`src/client/this-is-another-file-with-a-long-name-to-test-the-table-sizing.tsx\` | 0 B | 1.04 KB | ðŸ†• +1.04 KB
\`src/client/helpers/this-file-is-much-larger.ts\` | 1.35 KB | 13.01 KB | ðŸ“ˆ +11.66 KB (+865.77%)
\`src/client/this-file-grew-larger.tsx\` | 562 B | 762 B | ðŸ“ˆ +200 B (+35.59%)
\`src/client/helpers/this-file-grew-smaller.ts\` | 2.46 KB | 2.26 KB | ðŸ“‰ -200 B (-7.94%)
\`src/client/this-file-is-much-smaller.tsx\` | 11.67 KB | 10 B | ðŸ“‰ -11.66 KB (-99.92%)
\`src/client/routes.tsx\` | 2.72 KB | 0 B | ðŸ”¥ -2.72 KB (-100%)
\`src/client/this-file-will-be-deleted.ts\` | 1.46 KB | 0 B | ðŸ”¥ -1.46 KB (-100%)
\`src/client/render-memex.tsx\` | 1.04 KB | 0 B | ðŸ”¥ -1.04 KB (-100%)

</details>

</details>
"
`;

exports[`displays no module information when unchanged 1`] = `
"
**Changeset**

No files were changed"
`;

exports[`does not display module information when it does not exist 1`] = `""`;
